tests/unit/test.program.js

```diff
@@ -73,8 +73,8 @@ describe('program.Program', () => {
       shouldExitProgram: true,
     })
       .then(() => {
-        assert.equal(fakeProcess.exit.called, true);
-        assert.equal(fakeProcess.exit.firstCall.args[0], 1);
+        sinon.assert.calledOnce(fakeProcess.exit);
+        sinon.assert.calledWithMatch(fakeProcess.exit, 1);
```

Comment:
`assert.calledWithMatch()` is really only good for cases where you don't need to check all function arguments or if you only want to allow arbitrary object parameters in the value that you're checking. For all other cases you should use `assert.calledWith()`. 

As a general rule of thumb, first start with `assert.calledWith()` and if the test fails because the check is too strict (e.g. for comparing objects), consider switching to `assert.calledWithMatch()`.

In this case, the test should definitely use `assert.calledWith()` because it needs to check that `exit()` is called exactly with the number `1`. 

Here is a patch you can try:

```diff
diff --git a/src/program.js b/src/program.js
index 07559c5..fed0a93 100644
--- a/src/program.js
+++ b/src/program.js
@@ -161,7 +161,8 @@ export class Program {
       }
 
       if (this.shouldExitProgram) {
-        systemProcess.exit(1);
+        // $FLOW_IGNORE
+        systemProcess.exit(true);
       } else {
         throw error;
       }
```

This should cause the test to fail but when using `assert.calledWithMatch()` it passes since it's using a loose equality check (`==`).

As you can see, Flow helps us mitigate problems like this -- I was only using this as an example to show how the tests will need to avoid loose equality. The `assert.calledWith()` method will use strict equality (`===`).